publish.gradle

```diff
@@ -47,43 +47,55 @@ configure(subprojects.findAll({ it.name in ["persistence", "synchronization", "d
+                    def isMock = isFlavored && flavor.contains('Mock')
+                    if (isDebug || isMock) {
+                        return
                     }
+
+                    // Setup javadoc
+                    def optionalDebugAppendix = isDebug ? variant.getBuildType().name.capitalize() : ''
+                    def javaDocDestDir = file("${buildDir}/docs/javadoc ${isFlavored ? flavor + optionalDebugAppendix : ""}")
                     def javadoc = task("${variant.name}Javadoc", type: Javadoc) {
                         description "Generates Javadoc for ${variant.name}."
                         source = variant.javaCompileProvider.get().source
                         destinationDir = javaDocDestDir
-                        classpath += classpath + files(android.getBootClasspath().join(File.pathSeparator))
+                        classpath += files(android.getBootClasspath().join(File.pathSeparator))
                         classpath += files(configurations.compile)
                         options.links("http://docs.oracle.com/javase/7/docs/api/")
                         options.links("http://d.android.com/reference/")
-                        exclude '**/BuildConfig.java'
-                        exclude '**/R.java'
+                        exclude '**     /BuildConfig.java'
+                        exclude '**      /R.java'
                         failOnError false
                     }
+
+                    // Put javadoc and sources into jar
                     def javadocJar = task("${variant.name}JavadocJar", type: Jar, dependsOn: javadoc) {
                         description "Puts Javadoc for ${variant.name} in a jar."
                         archiveClassifier.set('javadoc')
                         from javadoc.destinationDir
                     }
+                    def sourceDirs = variant.sourceSets.collect {
+                        it.javaDirectories // includes Kotlin sources
+                    }
                     def sourcesJar = task("${variant.name}SourcesJar", type: Jar) {
                         description "Puts sources for ${variant.name} in a jar."
                         from sourceDirs
                         archiveClassifier.set('sources')
                     }
 
-                    "${project.name}${variant.name.capitalize()}${variant.getBuildType().name == 'release' ? '' : variant.getBuildType().name.capitalize()}"(MavenPublication) {
+                    // Set artifact name to [submodule, submoduleMovebis]
+                    def optionalMovebisFlavor = flavor.contains('movebis') ? 'movebis' : ''
+
+                    // Configure artifacts
+                    "${project.name}${variant.name.capitalize()}"(MavenPublication) {
                         groupId rootProject.ext.groupId
-                        artifactId "${project.name}${variantArtifactId.capitalize()}"
+                        artifactId "${project.name}${optionalMovebisFlavor.capitalize()}"
```

Comment:
The old code generated artifacts like:
- synchronizationCyfaceFullDebug
- persistencePersitenceDebug
etc.

My goal was to simplify the naming. This should be ok for the new API changing release (V5) I think, as long as there is a migration guide (task created: https://cyface.atlassian.net/browse/CY-5031)